DuckDuckGo/MainViewController+Automation.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  MainViewController+Automation.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+
+extension MainViewController {
+    
+    func executeScript(_ javaScriptString: String,
+                       args: [String: Any] = [:]) async -> Result<Any, any Error>? {
+        var result = await currentTab?.executeScript(javaScriptString, args: args)
+        return result! // TODO fix !
+    }
+
+    func currentUrl() -> String? {
```

Comment:
As commented elsewhere remove this, just access the webView url directly in the one place you need it